lib/red_cross/trackers/monitor_tracker.rb

```diff
@@ -3,12 +3,20 @@ module Trackers
     class MonitorTracker < RedCross::Trackers::Base
       attr_accessor  :client
 
-      def initialize(database = 'test', host = 'localhost', port = 8086, write_timeout = 0.05, read_timeout = 0.05, logger = false)
+      def initialize(database = 'test', host = 'localhost', port = 8086, write_timeout = 0.05, read_timeout = 0.05,
+                     logger = false, async = true)
         InfluxDB::Logging.logger = logger
+        async_options = async
+        if async_options == true
+          async_options = {
+              num_worker_threads: 1
+          }
+        end
+
         @client = InfluxDB::Client.new database,
                                        host:  host,
                                        port:  port,
-                                       async: true,
+                                       async: async_options,
```

Comment:
@alonbr are we sure that this is what we want? it seems to be exactly the opposite?

Reply:
@vladshub It seems like the dirty exit is related to the async param, in case it is false or the number of threads is one then it seems to prevent the dirty exits